Mathlib/RingTheory/LocalRing/MaximalIdeal/Basic.lean

```diff
@@ -58,6 +58,12 @@ instance : Unique (MaximalSpectrum R) where
   default := ⟨maximalIdeal R, maximalIdeal.isMaximal R⟩
   uniq := fun I ↦ MaximalSpectrum.ext_iff.mpr <| eq_maximalIdeal I.isMaximal
 
+/-- If the maximal spectrum of a ring is a singleton, then the ring is local. -/
+instance [Subsingleton (MaximalSpectrum R)] [Nonempty (MaximalSpectrum R)] : IsLocalRing R :=
```

Comment:
I personally think I would rather have the instances in the other direction, and this version as a lemma.

Because this spawns a search for instances of the pretty generic `Subsingleton` and `Nonempty` classes, which basically light up the entire TC graph.

The other direction is also syntax-driven, in the sense that a search for `SomeClass (complicated expr)` fires a search for `OtherClass (easier expr)`.